parser.hpp

```diff
@@ -97,21 +117,32 @@ CONSTEXPR CandyState decode_candy_state(char up_char)
 
 constexpr int candy_size = 3;
 constexpr int candy_type_offset = 1;
+constexpr int row_padding = 5;    // space, pipe, pipe, space, newline
+constexpr int noncandy_rows = 3;  // title, border, border
 
-constexpr auto parse_board_column_count(const constexpr_string_view& str)
+constexpr auto parse_board_parameters(auto&& str)
 {
-    const auto endline = find(str.cbegin(), str.cend(), '\n');
-    return (endline - str.cbegin()) / candy_size;
-}
+    BoardParameters ret{ 0, 0, 0, 0 };
 
-constexpr auto parse_board_row_count(const constexpr_string_view& str)
-{
-    int row_count = count(str.cbegin(), str.cend(), '\n');
+    auto gs_begin = str.cbegin();
+    auto gs_end = str.cend();
+    auto lefttop_corner = find(gs_begin + 1, gs_end, '\n');
```

Comment:
Ins't the lefttop_corner actually one offset after '\n'?

Reply:
True, renamed variables for "anchors"!